### PR TITLE
[scroll-animations-1] Remove -attachment, add timeline-scope

### PR DESCRIPTION
**Pull request by andruud**
#7759

This is specifically the model summarized in:
https://github.com/w3c/csswg-drafts/issues/7759#issuecomment-1527936739

### CODE REVIEW

**Comment by fantasai** [comment]
Heh. I actually have a partially completed fix for this on my system from yesterday also. :) We can merge the differences...

File: scroll-animations-1/Overview.bs [hunk 1]
```diff
@@ -385,78 +367,18 @@ spec: cssom-view-1; type: dfn;
...
-			If no such ancestor exists,
-			the [=attachment=] is treated as ''scroll-timeline-attachment/local''.
-	</dl>
-
 ### Scroll Timeline Shorthand: the 'scroll-timeline' shorthand ### {#scroll-timeline-shorthand}
 
 	<pre class='propdef shorthand'>
 	Name: scroll-timeline
-	Value: [ <<'scroll-timeline-name'>> [ <<'scroll-timeline-axis'>> || <<'scroll-timeline-attachment'>> ]? ]#
+	Value: [ <<'scroll-timeline-name'>> [ <<'scroll-timeline-axis'>> ]? ]#
```
Comment (fantasai) [inline scroll-animations-1/Overview.bs:374]: Need to remove the now-extraneous brackets.

File: scroll-animations-1/Overview.bs [hunk 2]
```diff
@@ -825,78 +732,66 @@ spec: cssom-view-1; type: dfn;
...
-### Attachment of a View Progress Timeline: the 'view-timeline-attachment' property ### {#view-timeline-attachment}
+### View Timeline Shorthand: the 'view-timeline' shorthand ### {#view-timeline-shorthand}
 
-	<pre class='propdef'>
-	Name: view-timeline-attachment
-	Value: [ local | defer | ancestor ]#
-	Initial: local
+	<pre class='propdef shorthand'>
+	Name: view-timeline
+	Value: [ <<'view-timeline-name'>> [ <<'view-timeline-axis'>> ]? ]#
```
Comment (fantasai) [inline scroll-animations-1/Overview.bs:739]: Same here: don't need the brackets anymore.

File: scroll-animations-1/Overview.bs [hunk 3]
```diff
@@ -825,78 +732,66 @@ spec: cssom-view-1; type: dfn;
...
-	Values have the following meanings:
+	ISSUE: Should it reset 'view-timeline-inset' also?
 
-	<dl dfn-type=value dfn-for=view-timeline-attachment>
-		<dt><dfn>local</dfn>
-		<dd>
-			This [=view progress timeline=] name
-			[=attaches=] to the corresponding [=view progress timeline=]
-			defined on this box.
+# Deferred Progress Timelines # {#deferred-timeline}
```
Comment (fantasai) [inline scroll-animations-1/Overview.bs:750]: Same top-level comment as earlier: I think I'd prefer to cast this as a name-scoping mechanism.

Also, this ultimately needs to go into css-animations-2, not here. Might make sense for us to just draft it there.

File: scroll-animations-1/Overview.bs [hunk 4]
```diff
@@ -825,78 +732,66 @@ spec: cssom-view-1; type: dfn;
...
+	with a matching name.
+	If more than one matching timeline exists,
+	or if no such timeline exists,
+	then the deferred timeline has no [=attached timeline=].
+
+## Naming a Deferred Progress Timeline: the 'timeline-scope' property ## {#timeline-scope}
+
+	<pre class='propdef'>
+	Name: timeline-scope
+	Value: none | [ <<dashed-ident>> ]#
```
Comment (fantasai) [inline scroll-animations-1/Overview.bs:786]: This is partially mixing up the dashed-ident change into the timeline-scope change, which leaves the spec in a weird inconsistent state.

File: web-animations-2/Overview.bs [hunk 5]
```diff
@@ -2250,13 +2250,14 @@ Update the attribute type for currentTime.
...
 
-Update the startTime and currentTime of the Animation interface as follows:
+Update the startTime, currentTime, and timeline of the Animation interface as follows:
 
 <pre class="idl">
 [Exposed=Window]
 partial interface Animation {
     attribute CSSNumberish?       startTime;
     attribute CSSNumberish?       currentTime;
+    attribute AnimationTimeline?  timeline;
```
Comment (fantasai) [inline web-animations-2/Overview.bs:2260]: Afaict this is an unrelated change, so it should go into its own issue/PR.

**Review by fantasai — CHANGES_REQUESTED** [review]
Some comments from comparing diffs...

**Comment by fantasai** [comment]
OK, I've pushed my version to https://github.com/w3c/csswg-drafts/pull/8906
Lmk what you think.

File: scroll-animations-1/Overview.bs (hunk 3 above)
Comment (andruud) [reply scroll-animations-1/Overview.bs:750]: Yeah, I'm not _actively_ trying to annoy you by going in circles. I was a definitely unsure what to do here, since the issue notes appear to show an unresolved situation where everyone except you resolved on this model https://github.com/w3c/csswg-drafts/issues/7759#issuecomment-1527936739 :upside_down_face:. But after conferring with Tab, they thought writing that model would be fine despite that.

Comment (fantasai) [reply scroll-animations-1/Overview.bs:750]: I think the version in https://github.com/w3c/csswg-drafts/pull/8906 is functionally saying the same thing, just describing it differently.
